src/util_i_targeting.nss

```diff
@@ -0,0 +1,894 @@
+/// @param sVarName The VarName associated with the target list.
+/// @param nIndex The index at which to delete the target data. If omitted,
+///     the first target on the list will be deleted.
+/// @returns The number of targets remaining on oPC's sVarName target list
+///     after deletion.
+int DeleteTargetingHookTarget(object oPC, string sVarName, int nIndex = 1);
+
+/// @brief Retrieve the target list object's internal index.
+/// @param oPC The PC object associated with the target list.
+/// @param sVarName The VarName associated with the target list.
+/// @param oObject The object to find on oPC's sVarName target list.
+int GetTargetingHookIndex(object oPC, string sVarName, object oTarget);
+
+/// @brief Delete target list target data by internal index.
+/// @param oPC The PC object associated with the target list.
+/// @param sVarName The VarName associated with the target list.
+/// @param nIndex The internal index of the target data to be deleted. This
+///     index can be retrieved from GetTargetingHookIndex().
+/// @returns The number of targets remaining on oPC's sVarName target list
+///     after deletion.
+int DeleteTargetingHookTargetByIndex(object oPC, string sVarName, int nIndex);
+
+// -----------------------------------------------------------------------------
+//                            Private Function Definitions
+// -----------------------------------------------------------------------------
+
+sqlquery _PrepareTargetingQuery(string s)
+{
+    return SqlPrepareQueryObject(GetModule(), s);
+}
+
+string _GetTargetingHookFieldData(int nHookID, string sField)
+{
+    string s =  "SELECT " + sField + " " +
+                "FROM targeting_hooks " +
+                "WHERE nHookID = @nHookID;";
+    sqlquery q = _PrepareTargetingQuery(s);
+    SqlBindInt(q, "@nHookID", nHookID);
+
+    return SqlStep(q) ? SqlGetString(q, 0) : "";
+}
+
+void _CreateTargetingDataTables(int bReset = FALSE)
+{
+    object oModule = GetModule();
+
+    if (bReset)
+    {
+        string sDropHooks = "DROP TABLE IF EXISTS targeting_hooks;";
+        string sDropTargets = "DROP TABLE IF EXISTS targeting_targets;";
+
+        sqlquery q;
+        q = _PrepareTargetingQuery(sDropHooks);   SqlStep(q);
+        q = _PrepareTargetingQuery(sDropTargets); SqlStep(q);
+
+        DeleteLocalInt(oModule, "TARGETING_INITIALIZED");
+        Warning(HexColorString("Targeting database tables have been dropped", COLOR_RED_LIGHT));
+    }
+
+    if (GetLocalInt(oModule, "TARGETING_INITIALIZED"))
+        return;
+
+    string sData = "CREATE TABLE IF NOT EXISTS targeting_hooks (" +
+        "nHookID INTEGER PRIMARY KEY AUTOINCREMENT, " +
+        "sUUID TEXT, " +
+        "sVarName TEXT, " +
+        "nObjectType INTEGER, " +
+        "nUses INTEGER default '1', " +
+        "sScript TEXT, " +
+        "UNIQUE (sUUID, sVarName));";
+
+    string sTargets = "CREATE TABLE IF NOT EXISTS targeting_targets (" +
```

Comment:
Would a join with `targeting_hooks` on `nHookID` be better than using `sUUID` and `sVarName`? Then you could use a foreign key to ensure the target table is cleaned up automatically when the hook is deleted. Or is it intended for the target list to stay around until manually deleted?

Reply:
Nevermind. Looks like I just didn't understand how the target list was meant to be used.

Reply:
Correct, the target lists are meant to be persistent, at least until the next server reset.